File: PFAnalisisAlgoritmosGo/utilidades/resultado.go
package utilidades

// ResultadoMet define la estructura para los resultados
type ResultadoMet struct {
	Nombre   string
	Casos    []CasoMet
	Lenguaje string
}

// ToJSON convierte el resultado a un formato JSON
func (resultado *ResultadoMet) ToJSON() map[string]interface{} {
	var casosJSON []map[string]interface{}
	for _, caso := range resultado.Casos {
		casosJSON = append(casosJSON, caso.ToJSON())
	}
	return map[string]interface{}{
		"nombre":   resultado.Nombre,
		"casos":    casosJSON,
		"lenguaje": resultado.Lenguaje,
	}
}

File: PFAnalisisAlgoritmosPython/utilidades/Resultado.py
class ResultadoMet:
    def __init__(self, nombre, casos, lenguaje):
        self.nombre = nombre
        self.casos = casos
        self.lenguaje = lenguaje
    def to_json(self):
        return {
            "nombre": self.nombre,
            "casos": [caso.to_json() for caso in self.casos],
            "lenguaje": self.lenguaje
        }